src/hotspot/share/prims/jvmtiEnvBase.hpp

```diff
@@ -499,36 +489,44 @@ class JvmtiHandshake : public Handshake {
   static void execute(JvmtiUnitedHandshakeClosure* hs_cl, jthread target);
 };
 
-class SetForceEarlyReturn : public JvmtiHandshakeClosure {
+class SetForceEarlyReturn : public JvmtiUnitedHandshakeClosure {
 private:
   JvmtiThreadState* _state;
   jvalue _value;
   TosState _tos;
 public:
   SetForceEarlyReturn(JvmtiThreadState* state, jvalue value, TosState tos)
-    : JvmtiHandshakeClosure("SetForceEarlyReturn"),
+    : JvmtiUnitedHandshakeClosure("SetForceEarlyReturn"),
      _state(state),
      _value(value),
      _tos(tos) {}
+  void doit(Thread *target, bool self);
```

Comment:
No need to use self, you might use _self from doit().

Reply:
Good suggestion, thanks.
The UpdateForPopTopFrameClosure::doit has the same issue. Fixed both now.